server/routes/index.js

```diff
@@ -656,22 +656,46 @@ module.exports = function Index ({ authenticationMiddleware }) {
     defaults,
     catchErrors(async (req, res) => {
       const {
-        params: { defendantId },
+        params: { courtCode, caseId, hearingId, defendantId },
         session,
-        path
+        path,
+        query: queryParams
       } = req
+
       const templateValues = await getCaseAndTemplateValues(req)
       templateValues.title = 'Review possible NDelius records'
-      const {
-        data: { defendantName }
-      } = templateValues
+
+      const { data: { defendantName } } = templateValues
       const response = await getMatchDetails(defendantId)
+
+      const matchingRecordsCount = response.offenderMatchDetails.length
```

Comment:
Do we ever check if this is zero? I know that to get here there must be a match as it's from a list but just wondering as it's a wild edge case.

Reply:
Also as we're not typescript can we check`response.offenderMatchDetails` is an array...